src/GafferScene/Light.cpp

```diff
@@ -100,6 +100,7 @@ void Light::affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs
 	if( input == defaultLightPlug() )
 	{
 		outputs.push_back( outPlug()->setNamesPlug() );
+		outputs.push_back( outPlug()->setPlug() );
```

Comment:
Seems like this is the correct fix if we want to be able to backport it, but perhaps is indicative of a hole in the ObjectSource API? In theory the Light class has no responsibility towards the `setPlug()` since that is meant to be dealt with in the ObjectSource base class. The subclasses are meant to only worry about `hashStandardSetNames()` and `computeStandardSetNames()`. This leads me to think we should have a matching `virtual bool affectsStandardSetNames( const Plug *input )` that subclasses implement and is called in `ObjectSource::affects()` - does that make sense? I don't think we need to implement this now, but a todo might be good, as I'm planning on standardising some similar `affects*()` methods as part of some optimisation work.